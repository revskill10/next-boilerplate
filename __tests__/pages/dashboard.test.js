/* eslint-env jest */

//import { shallow } from 'enzyme'
import App from '../../pages/dashboard.js'
import mockPage from '../helpers/mockPage'
import { createRender } from '@material-ui/core/test-utils';
import toJson from 'enzyme-to-json';
/*
describe('With Enzyme', () => {
  it('App shows "Hello world!"', () => {
    const app = shallow(<App />)

    expect(app.find('p').text()).toEqual('Hello World!')
  })
})
*/
describe('With Snapshot Testing', () => {
  let render;

  beforeEach(() => {
    render = createRender();
  });

  it('App shows "About"', () => {
    const component = mockPage(render)(App)
    expect(toJson(component)).toMatchSnapshot()
  })
})
