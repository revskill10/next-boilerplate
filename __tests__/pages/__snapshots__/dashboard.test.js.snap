// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing App shows "About" 1`] = `
Array [
  <header
    class="MuiPaper-root-20 MuiPaper-elevation4-26 MuiAppBar-root-11 MuiAppBar-positionStatic-15 MuiAppBar-colorDefault-17 Component-appBar-2"
  >
    <div
      class="MuiToolbar-root-47 MuiToolbar-regular-49 MuiToolbar-gutters-48"
    >
      <h6
        class="MuiTypography-root-51 MuiTypography-h6-68 MuiTypography-colorInherit-80 MuiTypography-noWrap-77 Component-toolbarTitle-3"
      >
        <a
          href="/"
          style="text-decoration:none;color:black"
        >
          TF
        </a>
      </h6>
      <h6
        class="MuiTypography-root-51 MuiTypography-h6-68 MuiTypography-colorInherit-80 MuiTypography-noWrap-77 Component-toolbarTitle-3"
      >
        <svg
          aria-hidden="true"
          class="sc-bwzfXH bqgLxe"
          fill="currentColor"
          focusable="false"
          height="40"
          size="40"
          viewBox="0 0 26 24"
          width="40"
        >
          <path
            d="M24.24 8l1.35-1.68C25.1 5.96 20.26 2 13 2S.9 5.96.42 6.32l12.57 15.66.01.02.01-.01L20 13.28V8h4.24z"
            fill-opacity="0.3"
          />
          <path
            d="M7.34 14.95L13 21.99V22v-.01l5.66-7.05C18.44 14.78 16.27 13 13 13s-5.44 1.78-5.66 1.95zM22 22h2v-2h-2v2zm0-12v8h2v-8h-2z"
          />
        </svg>
      </h6>
      <button
        class="MuiButtonBase-root-112 MuiButton-root-86 MuiButton-text-88 MuiButton-flat-91"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label-87"
        >
          <a
            class="Component-appBarLink-10"
            href="/dashboard"
          >
            dashboard_menu
          </a>
        </span>
        <span
          class="MuiTouchRipple-root-115"
        />
      </button>
      <button
        class="MuiButtonBase-root-112 MuiButton-root-86 MuiButton-text-88 MuiButton-flat-91"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label-87"
        >
          <a
            class="Component-appBarLink-10"
            href="/about"
          >
            about_menu
          </a>
        </span>
        <span
          class="MuiTouchRipple-root-115"
        />
      </button>
      <button
        class="MuiButtonBase-root-112 MuiButton-root-86 MuiButton-text-88 MuiButton-flat-91"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label-87"
        >
          <a
            class="Component-appBarLink-10"
            href="/pricing"
          >
            pricing_menu
          </a>
        </span>
        <span
          class="MuiTouchRipple-root-115"
        />
      </button>
      <button
        class="MuiButtonBase-root-112 MuiButton-root-86 MuiButton-text-88 MuiButton-flat-91"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label-87"
        >
          <a
            class="Component-appBarLink-10"
            href="/contact"
          >
            contact_menu
          </a>
        </span>
        <span
          class="MuiTouchRipple-root-115"
        />
      </button>
      <button
        class="MuiButtonBase-root-112 MuiButton-root-86 MuiButton-text-88 MuiButton-textPrimary-89 MuiButton-outlined-94 MuiButton-outlinedPrimary-95"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label-87"
        >
          <a
            href="/auth/google"
            style="text-decoration:none"
          >
            Login
          </a>
        </span>
        <span
          class="MuiTouchRipple-root-115"
        />
      </button>
    </div>
  </header>,
  <main>
    <div
      class="AutoGrid-root-122"
    >
      <div
        class="MuiGrid-container-123 MuiGrid-spacing-xs-24-147"
      >
        <div
          class="MuiGrid-item-124 MuiGrid-grid-xs-12-163"
        >
          <div
            class="ScrollableTabsButtonForce-root-220"
          >
            <header
              class="MuiPaper-root-20 MuiPaper-elevation4-26 MuiAppBar-root-11 MuiAppBar-positionStatic-15 MuiAppBar-colorDefault-17"
            >
              <div
                class="MuiTabs-root-221"
              >
                <div>
                  <div
                    style="width:100px;height:100px;position:absolute;top:-10000px;overflow:scroll;-ms-overflow-style:scrollbar"
                  />
                </div>
                <div
                  class="MuiTabs-flexContainer-222"
                >
                  <div
                    class="MuiTabScrollButton-root-230 MuiTabs-scrollButtons-227"
                  />
                  <div
                    class="MuiTabs-scroller-224 MuiTabs-scrollable-226"
                    role="tablist"
                    style="margin-bottom:0"
                  >
                    <div
                      class="MuiTabs-flexContainer-222"
                    >
                      <button
                        aria-selected="true"
                        class="MuiButtonBase-root-112 MuiTab-root-231 MuiTab-textColorPrimary-234 MuiTab-selected-236 MuiTab-labelIcon-232"
                        role="tab"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiTab-wrapper-239"
                        >
                          <svg
                            class="sc-htpNat jiJFSw"
                            fill="currentColor"
                            focusable="false"
                            height="40"
                            role="img"
                            size="40"
                            title="Accounts"
                            viewBox="0 0 24 24"
                            width="40"
                          >
                            <title>
                              Accounts
                            </title>
                            <path
                              d="M16.5 12c1.38 0 2.49-1.12 2.49-2.5S17.88 7 16.5 7a2.5 2.5 0 0 0 0 5zM9 11c1.66 0 2.99-1.34 2.99-3S10.66 5 9 5C7.34 5 6 6.34 6 8s1.34 3 3 3zm7.5 3c-1.83 0-5.5.92-5.5 2.75V19h11v-2.25c0-1.83-3.67-2.75-5.5-2.75zM9 13c-2.33 0-7 1.17-7 3.5V19h7v-2.25c0-.85.33-2.34 2.37-3.47C10.5 13.1 9.66 13 9 13z"
                            />
                          </svg>
                          <span
                            class="MuiTab-labelContainer-240"
                          >
                            <span
                              class="MuiTab-label-241"
                            >
                              Accounts
                            </span>
                          </span>
                        </span>
                        <span
                          class="TabIndicator-root-243 TabIndicator-colorPrimary-244 MuiTabs-indicator-229"
                        />
                        <span
                          class="MuiTouchRipple-root-115"
                        />
                      </button>
                      <button
                        aria-selected="false"
                        class="MuiButtonBase-root-112 MuiTab-root-231 MuiTab-textColorPrimary-234 MuiTab-labelIcon-232"
                        role="tab"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiTab-wrapper-239"
                        >
                          <svg
                            class="sc-bxivhb fplDdI"
                            fill="currentColor"
                            focusable="false"
                            height="40"
                            role="img"
                            size="40"
                            title="Reporting"
                            viewBox="0 0 512 512"
                            width="40"
                          >
                            <title>
                              Reporting
                            </title>
                            <path
                              d="M464 32H48C21.49 32 0 53.49 0 80v352c0 26.51 21.49 48 48 48h416c26.51 0 48-21.49 48-48V80c0-26.51-21.49-48-48-48zm-6 400H54a6 6 0 0 1-6-6V86a6 6 0 0 1 6-6h404a6 6 0 0 1 6 6v340a6 6 0 0 1-6 6zm-42-92v24c0 6.627-5.373 12-12 12H204c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h200c6.627 0 12 5.373 12 12zm0-96v24c0 6.627-5.373 12-12 12H204c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h200c6.627 0 12 5.373 12 12zm0-96v24c0 6.627-5.373 12-12 12H204c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h200c6.627 0 12 5.373 12 12zm-252 12c0 19.882-16.118 36-36 36s-36-16.118-36-36 16.118-36 36-36 36 16.118 36 36zm0 96c0 19.882-16.118 36-36 36s-36-16.118-36-36 16.118-36 36-36 36 16.118 36 36zm0 96c0 19.882-16.118 36-36 36s-36-16.118-36-36 16.118-36 36-36 36 16.118 36 36z"
                              fill="currentColor"
                            />
                          </svg>
                          <span
                            class="MuiTab-labelContainer-240"
                          >
                            <span
                              class="MuiTab-label-241"
                            >
                              Reporting
                            </span>
                          </span>
                        </span>
                        <span
                          class="MuiTouchRipple-root-115"
                        />
                      </button>
                      <button
                        aria-selected="false"
                        class="MuiButtonBase-root-112 MuiTab-root-231 MuiTab-textColorPrimary-234 MuiTab-labelIcon-232"
                        role="tab"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiTab-wrapper-239"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root-246"
                            focusable="false"
                            role="presentation"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M12 2c-4.97 0-9 4.03-9 9 0 4.17 2.84 7.67 6.69 8.69L12 22l2.31-2.31C18.16 18.67 21 15.17 21 11c0-4.97-4.03-9-9-9zm0 2c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.3c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                            />
                            <path
                              d="M0 0h24v24H0z"
                              fill="none"
                            />
                          </svg>
                          <span
                            class="MuiTab-labelContainer-240"
                          >
                            <span
                              class="MuiTab-label-241"
                            >
                              Student Registrations
                            </span>
                          </span>
                        </span>
                        <span
                          class="MuiTouchRipple-root-115"
                        />
                      </button>
                      <button
                        aria-selected="false"
                        class="MuiButtonBase-root-112 MuiTab-root-231 MuiTab-textColorPrimary-234 MuiTab-labelIcon-232"
                        role="tab"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiTab-wrapper-239"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root-246"
                            focusable="false"
                            role="presentation"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M0 0h24v24H0z"
                              fill="none"
                            />
                            <path
                              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 17h-2v-2h2v2zm2.07-7.75l-.9.92C13.45 12.9 13 13.5 13 15h-2v-.5c0-1.1.45-2.1 1.17-2.83l1.24-1.26c.37-.36.59-.86.59-1.41 0-1.1-.9-2-2-2s-2 .9-2 2H8c0-2.21 1.79-4 4-4s4 1.79 4 4c0 .88-.36 1.68-.93 2.25z"
                            />
                          </svg>
                          <span
                            class="MuiTab-labelContainer-240"
                          >
                            <span
                              class="MuiTab-label-241"
                            >
                              Item Four
                            </span>
                          </span>
                        </span>
                        <span
                          class="MuiTouchRipple-root-115"
                        />
                      </button>
                      <button
                        aria-selected="false"
                        class="MuiButtonBase-root-112 MuiTab-root-231 MuiTab-textColorPrimary-234 MuiTab-labelIcon-232"
                        role="tab"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiTab-wrapper-239"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root-246"
                            focusable="false"
                            role="presentation"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M0 0h24v24H0z"
                              fill="none"
                            />
                            <path
                              d="M17.21 9l-4.38-6.56c-.19-.28-.51-.42-.83-.42-.32 0-.64.14-.83.43L6.79 9H2c-.55 0-1 .45-1 1 0 .09.01.18.04.27l2.54 9.27c.23.84 1 1.46 1.92 1.46h13c.92 0 1.69-.62 1.93-1.46l2.54-9.27L23 10c0-.55-.45-1-1-1h-4.79zM9 9l3-4.4L15 9H9zm3 8c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2z"
                            />
                          </svg>
                          <span
                            class="MuiTab-labelContainer-240"
                          >
                            <span
                              class="MuiTab-label-241"
                            >
                              Item Five
                            </span>
                          </span>
                        </span>
                        <span
                          class="MuiTouchRipple-root-115"
                        />
                      </button>
                      <button
                        aria-selected="false"
                        class="MuiButtonBase-root-112 MuiTab-root-231 MuiTab-textColorPrimary-234 MuiTab-labelIcon-232"
                        role="tab"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiTab-wrapper-239"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root-246"
                            focusable="false"
                            role="presentation"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M15 3H6c-.83 0-1.54.5-1.84 1.22l-3.02 7.05c-.09.23-.14.47-.14.73v2c0 1.1.9 2 2 2h6.31l-.95 4.57-.03.32c0 .41.17.79.44 1.06L9.83 23l6.59-6.59c.36-.36.58-.86.58-1.41V5c0-1.1-.9-2-2-2zm4 0v12h4V3h-4z"
                            />
                          </svg>
                          <span
                            class="MuiTab-labelContainer-240"
                          >
                            <span
                              class="MuiTab-label-241"
                            >
                              Item Six
                            </span>
                          </span>
                        </span>
                        <span
                          class="MuiTouchRipple-root-115"
                        />
                      </button>
                      <button
                        aria-selected="false"
                        class="MuiButtonBase-root-112 MuiTab-root-231 MuiTab-textColorPrimary-234 MuiTab-labelIcon-232"
                        role="tab"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiTab-wrapper-239"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root-246"
                            focusable="false"
                            role="presentation"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M0 0h24v24H0V0z"
                              fill="none"
                            />
                            <path
                              d="M1 21h4V9H1v12zm22-11c0-1.1-.9-2-2-2h-6.31l.95-4.57.03-.32c0-.41-.17-.79-.44-1.06L14.17 1 7.59 7.59C7.22 7.95 7 8.45 7 9v10c0 1.1.9 2 2 2h9c.83 0 1.54-.5 1.84-1.22l3.02-7.05c.09-.23.14-.47.14-.73v-2z"
                            />
                          </svg>
                          <span
                            class="MuiTab-labelContainer-240"
                          >
                            <span
                              class="MuiTab-label-241"
                            >
                              Item Seven
                            </span>
                          </span>
                        </span>
                        <span
                          class="MuiTouchRipple-root-115"
                        />
                      </button>
                    </div>
                  </div>
                  <div
                    class="MuiTabScrollButton-root-230 MuiTabs-scrollButtons-227"
                  />
                </div>
              </div>
            </header>
            <div
              class="MuiTypography-root-51 MuiTypography-body1-60"
              style="padding:24px"
            >
              <p>
                ...
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </main>,
  <footer
    class="Component-footer-262 Component-layout-257"
  >
    <div
      class="MuiGrid-container-123 MuiGrid-spacing-xs-32-148 MuiGrid-justify-xs-space-evenly-144"
    >
      <div
        class="MuiGrid-item-124 MuiGrid-grid-xs-true-151"
      >
        <h6
          class="MuiTypography-root-51 MuiTypography-h6-68 MuiTypography-colorTextPrimary-83 MuiTypography-gutterBottom-78"
        >
          choose_language
        </h6>
        <div
          class="MuiInputBase-root-284 MuiInput-root-271 MuiInput-underline-275"
        >
          <div
            class="MuiSelect-root-264"
          >
            <div
              aria-haspopup="true"
              aria-pressed="false"
              class="MuiSelect-select-265 MuiSelect-selectMenu-268 MuiInputBase-input-294 MuiInput-input-279"
              role="button"
              tabindex="0"
            >
              language_en
            </div>
            <input
              type="hidden"
              value="en"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root-246 MuiSelect-icon-270"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-item-124 MuiGrid-grid-xs-true-151"
      >
        <h6
          class="MuiTypography-root-51 MuiTypography-h6-68 MuiTypography-colorTextPrimary-83 MuiTypography-gutterBottom-78"
        >
          Company
        </h6>
        <h6
          class="MuiTypography-root-51 MuiTypography-subtitle1-69 MuiTypography-colorTextSecondary-84"
        >
          Team
        </h6>
        <h6
          class="MuiTypography-root-51 MuiTypography-subtitle1-69 MuiTypography-colorTextSecondary-84"
        >
          History
        </h6>
        <h6
          class="MuiTypography-root-51 MuiTypography-subtitle1-69 MuiTypography-colorTextSecondary-84"
        >
          Contact us
        </h6>
        <h6
          class="MuiTypography-root-51 MuiTypography-subtitle1-69 MuiTypography-colorTextSecondary-84"
        >
          Locations
        </h6>
      </div>
      <div
        class="MuiGrid-item-124 MuiGrid-grid-xs-true-151"
      >
        <h6
          class="MuiTypography-root-51 MuiTypography-h6-68 MuiTypography-colorTextPrimary-83 MuiTypography-gutterBottom-78"
        >
          Features
        </h6>
        <h6
          class="MuiTypography-root-51 MuiTypography-subtitle1-69 MuiTypography-colorTextSecondary-84"
        >
          Cool stuff
        </h6>
        <h6
          class="MuiTypography-root-51 MuiTypography-subtitle1-69 MuiTypography-colorTextSecondary-84"
        >
          Random feature
        </h6>
        <h6
          class="MuiTypography-root-51 MuiTypography-subtitle1-69 MuiTypography-colorTextSecondary-84"
        >
          Team feature
        </h6>
        <h6
          class="MuiTypography-root-51 MuiTypography-subtitle1-69 MuiTypography-colorTextSecondary-84"
        >
          Developer stuff
        </h6>
        <h6
          class="MuiTypography-root-51 MuiTypography-subtitle1-69 MuiTypography-colorTextSecondary-84"
        >
          Another one
        </h6>
      </div>
      <div
        class="MuiGrid-item-124 MuiGrid-grid-xs-true-151"
      >
        <h6
          class="MuiTypography-root-51 MuiTypography-h6-68 MuiTypography-colorTextPrimary-83 MuiTypography-gutterBottom-78"
        >
          Resources
        </h6>
        <h6
          class="MuiTypography-root-51 MuiTypography-subtitle1-69 MuiTypography-colorTextSecondary-84"
        >
          Resource
        </h6>
        <h6
          class="MuiTypography-root-51 MuiTypography-subtitle1-69 MuiTypography-colorTextSecondary-84"
        >
          Resource name
        </h6>
        <h6
          class="MuiTypography-root-51 MuiTypography-subtitle1-69 MuiTypography-colorTextSecondary-84"
        >
          Another resource
        </h6>
        <h6
          class="MuiTypography-root-51 MuiTypography-subtitle1-69 MuiTypography-colorTextSecondary-84"
        >
          Final resource
        </h6>
      </div>
      <div
        class="MuiGrid-item-124 MuiGrid-grid-xs-true-151"
      >
        <h6
          class="MuiTypography-root-51 MuiTypography-h6-68 MuiTypography-colorTextPrimary-83 MuiTypography-gutterBottom-78"
        >
          Legal
        </h6>
        <h6
          class="MuiTypography-root-51 MuiTypography-subtitle1-69 MuiTypography-colorTextSecondary-84"
        >
          Privacy policy
        </h6>
        <h6
          class="MuiTypography-root-51 MuiTypography-subtitle1-69 MuiTypography-colorTextSecondary-84"
        >
          Terms of use
        </h6>
      </div>
    </div>
  </footer>,
]
`;
